controllers/idler/idler_controller_test.go

```diff
@@ -682,51 +682,89 @@ func TestAppNameTypeForControllers(t *testing.T) {
+			expectedAppType: "Deployment",
+		},
+	}
 
-		// second reconcile should delete pods and create notification
-		res, err = reconciler.Reconcile(context.TODO(), req)
-		//then
-		assert.NoError(t, err)
-		memberoperatortest.AssertThatIdler(t, idler.Name, cl).
-			HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
-		//check the notification is actually created
-		hostCl, _ := reconciler.GetHostCluster()
-		notification := &toolchainv1alpha1.Notification{}
-		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
-			Namespace: test.HostOperatorNs,
-			Name:      "alex-stage-idled",
-		}, notification)
-		require.NoError(t, err)
-		require.Equal(t, "alex@test.com", notification.Spec.Recipient)
-		require.Equal(t, "idled", notification.Labels[toolchainv1alpha1.NotificationTypeLabelKey])
-		require.Equal(t, "Pod", notification.Spec.Context["AppType"])
-		require.Equal(t, p.Name, notification.Spec.Context["AppName"])
+	for pt, tcs := range testpod {
+		t.Run(pt, func(t *testing.T) {
+			reconciler, req, cl, _ := prepareReconcile(t, idler.Name, getHostCluster, idler, nsTmplSet, mur)
+			idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
+			if pt == "Individual-Completed-Pod" || pt == "Individual-NonCompleted-Pod" {
+				preparePayloadsSinglePod(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond)
+			} else {
+				preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo, tcs.pcond)
+			}
+			// first reconcile to track pods
+			res, err := reconciler.Reconcile(context.TODO(), req)
+			assert.NoError(t, err)
+			assert.True(t, res.Requeue)
 
-	})
+			// second reconcile should delete pods and create notification
+			res, err = reconciler.Reconcile(context.TODO(), req)
+			//then
+			assert.NoError(t, err)
+
+			if pt == "Individual-Completed-Pod" {
+				memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+					HasConditions(memberoperatortest.Running())
+			} else {
+				memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+					HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
+			}
+			//check the notification is actually created
+			hostCl, _ := reconciler.GetHostCluster()
+			notification := &toolchainv1alpha1.Notification{}
+			err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
+				Namespace: test.HostOperatorNs,
+				Name:      "feny-stage-idled",
+			}, notification)
+
+			if pt == "Individual-Completed-Pod" {
+				require.EqualError(t, err, "notifications.toolchain.dev.openshift.com \"feny-stage-idled\" not found")
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, "feny@test.com", notification.Spec.Recipient)
+				require.Equal(t, "idled", notification.Labels[toolchainv1alpha1.NotificationTypeLabelKey])
+				require.Equal(t, tcs.expectedAppType, notification.Spec.Context["AppType"])
```

Comment:
What about checking the AppName?

Reply:
i forgot to add it for individual pod when i refactored the test case, added the check now for individual pod